app/models/form/lettings/pages/tenancy_length.rb

```diff
@@ -2,10 +2,10 @@ class Form::Lettings::Pages::TenancyLength < ::Form::Page
   def initialize(id, hsh, subsection)
     super
     @id = "tenancy_length"
-    @depends_on = [{ "tenancy" => 4 }, { "tenancy" => 6 }, { "tenancy" => 3 }]
+    @depends_on = [{ "tenancy_type_fixed_term?" => true }]
```

Comment:
@natdeanlewissoftwire 
if you're happy this change covers 2034 shall we close that PR and comment on that JIRA ticket and this one that testing for that should be covered in testing this ticket?